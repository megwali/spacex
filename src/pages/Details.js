import styled from 'styled-components';
import useLaunchList from '../hooks/useLaunchList';

const Card = styled.div`
  border: 1px solid lightgrey;
  padding: 10px;
`;

const RocketDetails = styled.p`
  color: lightgrey;
  font-size: 14px;
  font-weight: 500;
`;

const RocketName = styled.p`
  color: black;
  font-size: 18px;
  font-weight: 600;
`;

const Details = ({ id }) => {
  const [launch] = useLaunchList(id);
  const { fairings, rocket_name, rocket_type } = launch?.rocket || {};

  return !launch ? 'loading' : (
    <Card>
      <RocketName>Rocket name: {rocket_name}</RocketName>
      <RocketDetails>Rocket type: {rocket_type}</RocketDetails>
      <RocketDetails>Reused: {fairings?.reused ? 'Yes' : 'No'}</RocketDetails>
    </Card>
  );
};

export default Details;
