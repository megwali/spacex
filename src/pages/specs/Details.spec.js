import { create } from 'react-test-renderer';
import Details from '../Details';

describe('Details', () => {
  it ('renders', () => {
    const component = create(<Details />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
