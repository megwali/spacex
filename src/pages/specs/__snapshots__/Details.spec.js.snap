// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders 1`] = `"loading"`;
